cli/src/complete.rs

```diff
@@ -28,19 +29,49 @@ use crate::config::default_config;
 use crate::config::LayeredConfigs;
 use crate::ui::Ui;
 
+/// A helper function for various bookmark completer functions. It returns a
+/// template that formats the bookmarks in a standardized way, such that it can
+/// be parsed by `bookmark_line_to_name_and_help`.
+fn bookmark_template_from_filter(filter: &str) -> String {
+    format!(
+        r#"if({filter},
+            name ++ if(remote, "@" ++ remote) ++ "\t" ++
+            if(normal_target,
+                normal_target.change_id().shortest() ++ ": " ++
+                if(normal_target.description(),
+                    normal_target.description().first_line(),
+                    "(no description set)",
+                ),
+                "(conflicted bookmark)",
+            ) ++ "\n"
```

Comment:
It should be okay to assume that the summary template contains no newlines. If it contained newlines, various outputs would be broken.

I don't know if people want to customize the completion output. I just thought the information to be wanted here would be basically the same as `jj bookmark list`.

Reply:
> I just thought the information to be wanted here would be basically the same as `jj bookmark list`.

It feels a little verbose to me. If the help text is small enough, shells can distribute suggestions in a table to fit more on screen. So we should avoid too much irrelevant information. For example, I usually don't care about commit ids. Showing them in the help text doesn't give me any information if that's the bookmark I wanted to pick.

Also, `format_commit_summary_with_refs` contains _short_ ids, not _shortest_ ids. This is fine for jj output, which is colored nicely so the _shortest_ id is still identifiable. But the help text in shell completions loses this coloring. I think the shortest id is more valuable to show than the short one.

Reply:
>  I usually don't care about commit ids

That's fair.

> is colored nicely so the shortest id is still identifiable. But the help text in shell completions loses this coloring.

I assumed the help text can be colorized because its type is `StyledStr`.

Reply:
Yeah, fish doesn't respect the styling. So we shouldn't rely on it as a means to display information.

Reply:
Never mind. Appears that escape chars are stripped from hint by clap.